src/main/scala/org/hammerlab/guacamole/reads/Read.scala

```diff
@@ -75,6 +75,9 @@ trait Read {
   /** Whether read is from a paired-end library */
   val isPaired: Boolean
 
+  /** Whether the read has associated reference data */
+  val hasMdTag: Boolean
```

Comment:
all the abstract `val`s in this `trait` don't make me feel great. Is there rhyme or reason to them? @arahuja @timodonnell

cf. http://stackoverflow.com/a/19642301/544236